core/java/iesi-core/src/main/java/io/metadew/iesi/script/execution/instruction/data/text/TextSubstring.java

```diff
@@ -14,24 +15,43 @@
     private final Pattern INPUT_PARAMETER_PATTERN = Pattern.compile("(?<" + FIRST_OPERATOR + ">.+)," +
             "\\s*(?<" + SECOND_OPERATOR + ">-?\\d+)\\s*,\\s*(?<" + THIRD_OPERATOR + ">-?\\d+)\\s*");
 
+    private final Pattern INPUT_PARAMETER_PATTERN_TWO_ARGUMENTS = Pattern.compile("(?<" + FIRST_OPERATOR + ">.+)," +
+            "\\s*(?<" + SECOND_OPERATOR + ">-?\\d+)\\s*");
+
     @Override
     public String generateOutput(String parameters) {
         Matcher inputParameterMatcher = INPUT_PARAMETER_PATTERN.matcher(parameters);
-        if (!inputParameterMatcher.find()) {
-            throw new IllegalArgumentException(MessageFormat.format("Illegal arguments provided to " + this.getKeyword() + ": {0}", parameters));
-        } else {
+        Matcher inputParameterMatcherTwoArguments = INPUT_PARAMETER_PATTERN_TWO_ARGUMENTS.matcher(parameters);
+
+        if (inputParameterMatcher.find()) {
             String text = inputParameterMatcher.group(FIRST_OPERATOR);
             int start = Integer.parseInt(inputParameterMatcher.group(SECOND_OPERATOR));
             int end = Integer.parseInt(inputParameterMatcher.group(THIRD_OPERATOR));
-            if (start < 0) {
-                start = text.length() + start;
-            }
-            if (end < 0) {
-                end = text.length() + end;
-            }
-            verifyArguments(text, start, end);
-            return text.substring(start, end);
+
+            if (start < 0)
+                start += text.length()+1;
```

Comment:
you do plus 1 and in line 37 and 38, you do minus 1. Why first add 1 to later on subtract 1?

Reply:
hum, I've seen on the previous unit test that the last developer made you were expected a result when the user introduce an negative value => assertEquals("sts", textSubstring.generateOutput("teststring, -8, -6"));
so in order to get the same "result",  I adjusted the code when the user enter a negative value, and I don't know why the last programmer expected this result whit those value?

I think I should deleted this part but I wanted to know your position? should I?